utils/test/test_framework.h

```diff
@@ -1,9 +1,35 @@
 #ifndef TEST_FRAMEWORK_H_
 #define TEST_FRAMEWORK_H_
 
+#include <stdio.h>
+
 #define SUCCESS 0
 #define FAIL -1
-#define ASSERT(expr) if(!(expr)) return FAIL
-#define ASSERT_IS_NONE(var) if(var != NULL) return FAIL
+
+#ifdef TEST_NO_PRINT_
+
+#define ASSERT(expr)                                                           \
+    if (!(expr))                                                               \
+        return FAIL;
+
+#define ASSERT_IS_NONE(var)                                                    \
+    if (var != NULL)                                                           \
+        return FAIL;
+
+#else /*TEST_NO_PRINT*/
+
+#define ASSERT(expr)                                                           \
+    if (!(expr)) {                                                             \
+        printf("Assert failed on line %d\n", __LINE__);                        \
+        return FAIL;                                                           \
+    }
+
+#define ASSERT_IS_NONE(var)                                                    \
+    if (var != NULL) {                                                         \
+        printf("Assert failed on line %d\n", __LINE__);                        \
```

Comment:
Could change to `Assert is none failed on line %d\n`

Reply:
Is "ASSERT_IS_NULL failed on line %d\n" alright?